### PR TITLE
Utility

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added PageNotFound
- Added Navlink
- Added Favicon
- Added Document title

### CODE REVIEW

File: src/Frontend/Pages/Home/Header/index.css [hunk 1]
```diff
@@ -87,12 +87,13 @@ a {
...
+  color: var(--white);
 }
 .cart > a {
   color: var(--white);
   white-space: nowrap;
 }
-.cart > :nth-child(1) {
+.icon {
   color: var(--white);
   /* font-weight: 550;*/
```
Comment (Prabhav-Sharma) [inline src/Frontend/Pages/Home/Header/index.css:98]: Can remove unneeded comments

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/Frontend/Pages/Home/Search/index.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { searchProducts, useProducts } from "../../../Context/products";
+import "./index.css";
+
+function Index() {
+  const { dispatch, searchedProducts } = useProducts();
+
+  const getSearchResults = async (query) =>
+    dispatch({ ...(await searchProducts(query.toLowerCase())) });
+
+  const debounce = (func, timeout = 250) => {
```
Comment (ArjunGTX) [inline src/Frontend/Pages/Home/Search/index.jsx:14]: Is this debounce function tested? Are you sure you don't need to memoize this function?

**Review by ArjunGTX — APPROVED** [review]
Tried the deploy preview. Awesome UI. Noticed one flaw: There is no constraints for password. for example I created an account with password `abc`. It would be better if you could include some validation

**Review by SukanyaSen530 — APPROVED** [review]

File: src/Frontend/Pages/Home/Search/index.jsx (hunk 2 above)
Comment (rehanmohiuddin) [reply src/Frontend/Pages/Home/Search/index.jsx:14]: it's a simple debounce function tried to get my hand's dirty next iteration i will do memoization
Brute -> Better -> Best
And For Login Validation i have ignored wantedly as there are no marks for auth so thought i will do after submission

**Pull request merged**